extensions/Gemini/python/src/aiconfig_extension_gemini/Gemini.py

```diff
@@ -111,11 +118,58 @@ async def serialize(
         )
         await ai_config.callback_manager.run_callbacks(event)
 
-        raise Exception("Not Implemented")
+        # Extract model name from data
+
+        # Reform data object to be compatible with AIConfig Data Model (everything minus the prompt input and output)
+
+        # Parse into prompts. If no conversation history, return a single prompt
+
+        model_metadata = ai_config.get_model_metadata(data, self.model.model_name)
+
+        contents = data.get("contents", None)
+        prompts = []
+        is_str = isinstance(contents, str)
+        is_list_of_strings = all(isinstance(item, str) for item in contents) if isinstance(contents, list) else False
+        is_role_dict = isinstance(contents, dict) and "role" in contents and "parts"
+        is_multi_turn = isinstance(contents, list) and all(isinstance(item, dict) and "role" in item and "parts" in item for item in contents)
```

Comment:
Nit: add doc of API: https://ai.google.dev/tutorials/python_quickstart#multi-turn_conversations